lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,702 @@
+        private const int TotalNumberOfTasks = 3;
+
+        /// <summary>
+        /// Tests valid Add task cases
+        /// </summary>
+        [Fact]
+        public void TestValidAdd()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            Assert.Equal(TotalNumberOfTasks, taskManager.NumberOfTasks);
+        }
+
+        /// <summary>
+        /// Tests reset tasks in the TaskManager
+        /// </summary>
+        [Fact]
+        public void TestReset()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.Reset();
+            Assert.Equal(0, taskManager.NumberOfTasks);
+            Assert.Equal(0, taskManager.NumberOfAppError());
+        }
+
+        /// <summary>
+        /// Tests SubmitTasks after adding all the tasks to the TaskManager
+        /// </summary>
+        [Fact]
+        public void TestSubmitTasks()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+            Assert.True(taskManager.AreAllTasksInState(TaskState.TaskSubmitted));
+        }
+
+        /// <summary>
+        /// Tests SubmitTask with a missing mapper task
+        /// </summary>
+        [Fact]
+        public void TestMissingMapperTasksSubmit()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(0));
+            taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+
+            Action submit = () => taskManager.SubmitTasks();
+            Assert.Throws<IMRUSystemException>(submit);
+        }
+
+        /// <summary>
+        /// Tests SubmitTask with missing master task
+        /// </summary>
+        [Fact]
+        public void TestMissingMasterTaskSubmit()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+            taskManager.AddTask(MapperTaskIdPrefix + 2, MockConfig(), CreateMockActiveContext(2));
+
+            Action submit = () => taskManager.SubmitTasks();
+            Assert.Throws<IMRUSystemException>(submit);
+        }
+
+        /// <summary>
+        /// Tests adding all mapper tasks without master task
+        /// </summary>
+        [Fact]
+        public void NoMasterTask()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+            taskManager.AddTask(MapperTaskIdPrefix + 2, MockConfig(), CreateMockActiveContext(1));
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 3, MockConfig(), CreateMockActiveContext(3));
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests adding two master tasks
+        /// </summary>
+        [Fact]
+        public void TwoMasterTasks()
```

Comment:
This should be merged with `DuplicatedTaskIdInAdd` test (they cover the same scenario)